community/blog/2021-07 The power of Arcus secret store/README.md

```diff
@@ -0,0 +1,149 @@
+
+    // Built-in secret providers.
+    var environmentVariables = new EnvironmentVariableSecretProvider();
+    var cachedEnvironmentVariables = new CachedSecretProvider(environmentVariables, cacheConfiguration);
+
+    // Custom secret providers.
+    var windowsRegistry = new WindowsRegistrySecretProvider();
+    var cachedWindowsRegistry = new CahcedSecretProvider(windowsRegistry, CacheConfiguration.Default);
+});
+```
+
+See also our [asynchrounous cache invalidation](https://background-jobs.arcus-azure.net/features/security/auto-invalidate-secrets) to automatically invalidate cached Azure Key Vault secrets. 
+
+## Plug & play
+
+Our secret store is very easily to set up. All it requires is a single call to register the store `ConfigureSecretStore`, after that you can immediately use our `ISecretProvider` interface abstraction throughout your application.
+
+```csharp
+using Arcus.Security.Core;
+
+[ApiController]
+public class OrderController : ControllerBase
+{
+    private readonly ISecretProvider _secretStore;
+
+    public OrderController(ISecretProvider secretStore)
+    {
+        _secretStore = secretStore;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> Orders()
+    {
+        string secret = await _secretStore.GetRawSecretAsync("MySecret");
+    }
+}
+```
+
+We are very flexible in the registration of the secret providers. We allow all kinds of combinations and even multiple registrations of the same type. Imagine two Azure Key Vaults with public and user secrets.
+
+```csharp
+using Microsoft.Extensions.Hosting;
+
+.ConfigureSecretStore((config, stores) =>
+{
+    stores.AddAzureKeyVaultWithManagedServiceIdentity("https://public-secrets.vault.azure.net")
+          .AddAzureKeyVaultWithManagedServiceIdentity("https://user-secrets.vault.azure.net"); 
+})
+```
+
+> Note that with our [named secret providers feature](https://security.arcus-azure.net/features/secret-store/named-secret-providers), you can retrieve a single secret provider instead of the combination. This can be useful in a scenario like this.
+
+## Designed for security
+
+Besides all the user-friendly features, there're also security reasons why the Arcus secret store is a recommended choice. We made sure that all the secret providers are registered in one single place and are being contacted via one single access point. This is much safer and easier to maintain then writing multiple service integrations yourself, scattered across the application. 
+
+It's also safer than using the general [ASP.NET Core configuration](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/) approach because we explicitly made the distinction between **data** and **secrets**. Many vulnerabilities gets introduced when secrets are seeen as data and are included in logs, for example. Leakage of sensitive information is a real risk in this case. Or when expired secrets doesn't get transient handling upon retrieval.
```

Comment:
```suggestion

While [.NET's configuration](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/) approach feels very similar, it does not make an explicit distinction between **configuration** and **secrets**.

Many vulnerabilities get introduced when secrets are seen as data and are included in logs, for example. Leakage of sensitive information is a real risk in this case. Or when expired secrets don't get transient handling upon retrieval.

**With Arcus secret store, you are making secrets a first-class citizen in .NET Core** to ensure they are handled accordingly.

We make sure that all the secret providers are registered in a central place and are being contacted via the secret store. This is much safer and easier to maintain since it is easy to get an overview of the various secret sources that the application depends on.

Lastly, there is a lot less work for end-users since they no longer have to write multiple service integrations themselves, scattered across the application, but instead rely on a variety of sources.
```